backend/app/api/endpoints/mcp_providers.py

```
@@ -0,0 +1,134 @@
+        user_update = UserUpdate(preferences=existing_prefs)
+        user_service.update_current_user(
+            db=db,
+            user=current_user,
+            obj_in=user_update,
+        )
+
+        return MCPProviderKeysResponse(
+            success=True,
+            message="MCP provider keys updated successfully",
+        )
+    except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)